### PR TITLE
docs: update user guide/references for workload identity federation

### PR DESCRIPTION
**Pull request by bojeil-google**
Updates user guide to document using external account credentials with ADC or explicitly in various platforms.
Regenerates references for all classes and utilities related to this feature.

### CODE REVIEW

File: docs/reference/google.auth.credentials_async.rst [hunk 1]
```diff
@@ -1,7 +0,0 @@
-google.auth.credentials\_async module
```
Comment (busunkim96) [inline docs/reference/google.auth.credentials_async.rst:1]: @crwilcox Should we keep the RSTs for the async files or is it alright to delete them for now?

Comment (crwilcox) [reply docs/reference/google.auth.credentials_async.rst:1]: Not sure I understand the 'for now' here? I would assume we still want them?

Comment (busunkim96) [reply docs/reference/google.auth.credentials_async.rst:1]: The way `docgen` is set up it seems to try to wipe these. Opened https://github.com/googleapis/google-auth-library-python/issues/685 to track.

**Review by busunkim96 — APPROVED** [review]

**Comment by busunkim96** [comment]
CLAs were granted when changes were merged to the main branch.

**Pull request merged**